src/partition/repart.c

```diff
@@ -1417,7 +1435,9 @@ static int context_read_definitions(
                 if (!p->definition_path)
                         return log_oom();
 
-                r = partition_read_definition(p, *f);
+                p->drop_in_files = strv_copy(drop_in_files);
```

Comment:
You can replace `strv_copy()` with `TAKE_PTR()`, this will assign `drop_in_files` to `p->drop_in_files` and then set `drop_in_files` to `NULL`. This makes sure that the strv is not freed by `_cleanup_strv_free_`.